src/mlpack/methods/reinforcement_learning/replay/episodic_replay.hpp

```diff
@@ -0,0 +1,242 @@
+/**
+ * @file episodic_replay.hpp
+ * @author Narayanan E R
+ *
+ * This file is an implementation of episodic experience replay.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_REPLAY_EPISODIC_REPLAY_HPP
+#define MLPACK_METHODS_RL_REPLAY_EPISODIC_REPLAY_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace rl {
+
+/**
+ * Implementation of episodic experience replay.
+ *
+ * Each episode of interactions between the agent and the
+ * environment will be saved to a memory buffer. When necessary,
+ * we can simply sample previous episodes or the most recent episode
+ * from the buffer to train the agent.
+ *
+ * @tparam EnvironmentType Desired task.
+ */
+
+template <typename EnvironmentType>
+class EpisodicReplay
+{
+ public:
+  //! Convenient typedef for action.
+  using ActionType = typename EnvironmentType::Action;
+
+  //! Convenient typedef for state.
+  using StateType = typename EnvironmentType::State;
+
+  /**
+  * Construct an instance of episode experience replay class.
+  */
+  EpisodicReplay():
+      capacity(0),
+      position(0),
+      maxEpisodeLen(0),
+      full(false)
+  { /* Nothing to do here. */ }
+
+  /**
+  * Construct an instance of episode experience replay class.
+  *
+  * @param capacity Total memory size in terms of number of episodes.
```

Comment:
```suggestion
  * @param capacity Maximum number of episodes.
```
Perhaps this would be more clear?